reference/constraints/_payload-option.rst.inc

```diff
@@ -0,0 +1,6 @@
+payload
+~~~~~~~
+
+**type**: ``mixed`` **default**: ``null``
+
+This option can be used to attach arbitrary domain-specific data to a constraint.
```

Comment:
I do think that's very usefull. The current description doesn't add much information than what you could already know when reading the option name. I call those things the "no-shit-sherlock-comments" :wink: (just like the `Gets xxx` PHPdoc for getters...).

What we should describe imo:
* *Why was this option added, why would someone use this?*
* *What information shouldn't you put in here.* Since it's such a general option, I expect people to put things in here that don't belong there, we should clearly tell them for what type of things this option is created.
* *How to work with the option?* You can set the option, fine, but what happends after it? Is it shown in the violation message, can I access it with the errors, etc?

Looking at [JSR-303](http://beanvalidation.org/1.0/spec/#constraintsdefinitionimplementation-constraintdefinition-payload) (where this option is specified), one use-case is to add the "severity" (or importance) of the error. E.g., "error", "notice" or "warning".

This option isn't used by the Validator component or any other Symfony component. You can use it yourself. E.g:

```php
$violations = $validator->validate($documentationPage);

foreach ($violations as $violation) {
    $severity = $violation->getConstraint()->payload['severity'];
}
```

(please note that `Violation#getConstraint()` was introduced in Symfony 2.6)